cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CreateExecutableTask.java

```diff
@@ -158,6 +160,26 @@ public void execute(Project project) {
             }
         }
 
+        if (project.buildOptions().optimizeCodegen()) {
+            Path relativePathToExecutable = currentDir.relativize(executablePath);
+            String relativePathToExecutableString =
+                    relativePathToExecutable.toString().replace(".jar", "_OPTIMIZED.jar");
+
+            String executablePathString =
+                    executablePath.toString().replace(".jar", "_OPTIMIZED.jar");
```

Comment:
Can have constants declared here

Reply:
Conflicted on whether to change the name of the optimized executable jar at all. Will update here on the decision.

Reply:
We decided to emit two executable JARs.

1. BIR level optimized executable JAR (will contain `_BIR_OPTIMIZED` suffix).
2. BIR level + native dependency optimized JAR (will contain `_OPTIMZIED` suffix).